client/allocrunner/taskrunner/logmon_hook.go

```diff
@@ -59,10 +61,17 @@ func newLogMonHook(tr *TaskRunner, logger hclog.Logger) *logmonHook {
 	return hook
 }
 
-func newLogMonHookConfig(taskName, logDir string) *logmonHookConfig {
+func newLogMonHookConfig(taskName string, logCfg *structs.LogConfig, logDir string) *logmonHookConfig {
 	cfg := &logmonHookConfig{
-		logDir: logDir,
+		logDir:  logDir,
+		enabled: logCfg.Enabled,
 	}
+	if !logCfg.Enabled {
```

Comment:
Took me a second to remember exactly how all of this gets wired up.

```suggestion

  // If logging is disabled configure task's stdout/err to point to devnull
	if !logCfg.Enabled {
```